Build failure: vimPlugins.LanguageClient-neovim

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

```
$ NIXPKGS_ALLOW_BROKEN=1 nix-build -A vimPlugins.LanguageClient-neovim
```

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

https://hydra.nixos.org/job/nixpkgs/trunk/vimPlugins.LanguageClient-neovim.x86_64-linux

### Relevant log output

```console
$ NIXPKGS_ALLOW_BROKEN=1 nix-build -A vimPlugins.LanguageClient-neovim
[...]
error[E0119]: conflicting implementations of trait `Trait` for type `(dyn Send + Sync + 'static)`
  --> /build/LanguageClient-neovim-bin-0.1.161-vendor/traitobject-0.1.0/src/impls.rs:72:1
   |
71 | unsafe impl Trait for ::std::marker::Send + Sync { }
   | ------------------------------------------------ first implementation here
72 | unsafe impl Trait for ::std::marker::Send + Send + Sync { }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `(dyn Send + Sync + 'static)`

error[E0119]: conflicting implementations of trait `Trait` for type `(dyn Send + Sync + 'static)`
  --> /build/LanguageClient-neovim-bin-0.1.161-vendor/traitobject-0.1.0/src/impls.rs:73:1
   |
71 | unsafe impl Trait for ::std::marker::Send + Sync { }
   | ------------------------------------------------ first implementation here
72 | unsafe impl Trait for ::std::marker::Send + Send + Sync { }
73 | unsafe impl Trait for ::std::marker::Sync + Send { }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `(dyn Send + Sync + 'static)`

error[E0119]: conflicting implementations of trait `Trait` for type `(dyn Send + Sync + 'static)`
  --> /build/LanguageClient-neovim-bin-0.1.161-vendor/traitobject-0.1.0/src/impls.rs:75:1
   |
71 | unsafe impl Trait for ::std::marker::Send + Sync { }
   | ------------------------------------------------ first implementation here
...
75 | unsafe impl Trait for ::std::marker::Sync + Send + Sync { }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `(dyn Send + Sync + 'static)`

   Compiling same-file v1.0.6
For more information about this error, try `rustc --explain E0119`.
error: could not compile `traitobject` (lib) due to 3 previous errors
```

### Additional context

It seems upstream activity is low. E.g. https://github.com/autozimu/LanguageClient-neovim/issues/1265

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.35, NixOS, 25.05 (Warbler), 25.05.git.88983d4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - nixpkgs: `/nix/store/6s12fy3hizm5jipaz62il5f40f0v9mzb-nixpkgs-source`


### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/editors/vim/plugins/non-generated/LanguageClient-neovim/default.nix
+++ b/pkgs/applications/editors/vim/plugins/non-generated/LanguageClient-neovim/default.nix
@@ -17,7 +17,11 @@ let
     pname = "LanguageClient-neovim-bin";
     inherit version src;
 
-    cargoHash = "sha256-1tfeowqvjEjMXIfrhr388YhlZrk3ns+Y/2odQnkLw7k=";
+    cargoPatches = [
+      ./traitobject.patch
+    ];
+
+    cargoHash = "sha256-43alR84MktYTmsKeUMm4gK8AjUIkGqcsuFeQPusBKD0=";
   };
 in
 vimUtils.buildVimPlugin {
@@ -47,7 +51,5 @@ vimUtils.buildVimPlugin {
     homepage = "https://github.com/autozimu/LanguageClient-neovim/";
     changelog = "https://github.com/autozimu/LanguageClient-neovim/releases/tag/${src.tag}";
     license = lib.licenses.mit;
-    # Rust build error
-    broken = true;
   };
 }

--- a/pkgs/applications/editors/vim/plugins/non-generated/LanguageClient-neovim/traitobject.patch
+++ b/pkgs/applications/editors/vim/plugins/non-generated/LanguageClient-neovim/traitobject.patch
@@ -1,0 +1,15 @@
+--- a/Cargo.lock
++++ b/Cargo.lock
+@@ -1026,9 +1028,9 @@ dependencies = [
+
+ [[package]]
+ name = "traitobject"
+-version = "0.1.0"
++version = "0.1.1"
+ source = "registry+https://github.com/rust-lang/crates.io-index"
+-checksum = "efd1f82c56340fdf16f2a953d7bda4f8fdffba13d93b00844c25572110b26079"
++checksum = "04a79e25382e2e852e8da874249358d382ebaf259d0d34e75d8db16a7efabbc7"
+
+ [[package]]
+ name = "treediff"
+

